kafkaesk/ext/logging/handler.py

```diff
@@ -119,16 +143,53 @@ def _initialize_model(self) -> None:
         except kafkaesk.app.SchemaConflictException:
             pass
 
+    def _format_base_log(self, record: PydanticLogRecord) -> Dict[str, Any]:
+        if record.exc_text is None and record.exc_info:
+            record.exc_text = _formatter.formatException(record.exc_info)
+            try:
+                record.exc_type = record.exc_info[0].__name__  # type: ignore
+            except (AttributeError, IndexError):
+                ...
+
+        if record.stack_info:
+            record.stack_text = _formatter.formatStack(record.stack_info)
+
+        service_name = "unknown"
+        hostname = socket.gethostname()
+        if hostname.count("-") > 1:
+            # detect kubernetes service host
+            service_name = "-".join(hostname.split("-")[:-2])
```

Comment:
This probably works for most pods, but it could end up empty on say a stateful set that looks like `prometheus-0`, should you add a check that you have at least 1 part?